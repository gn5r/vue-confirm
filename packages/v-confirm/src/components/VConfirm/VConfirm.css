.transition-enter-active {
  transition: opacity 0.2s ease-in;
}

.transition-leave-active {
  transition: opacity 0.3s ease-out;
}

.transition-enter,
.transition-leave-to {
  opacity: 0;
}

.v-confirm {
  position: fixed;
  z-index: 200;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  outline: none;
}

.v-confirm > * {
  width: 100%;
}

.v-confirm--animated > .v-confirm__content {
  animation: bounce 0.2s;
}

@keyframes bounce {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.05);
  }
  100% {
    transform: scale(1);
  }
}

.v-confirm__content {
  background-color: #eee;
  border-radius: 3px;
  box-shadow: 1px 2px 3px 4px rgba(12, 12, 12, 0.2);
  max-width: 95%;
}

.v-confirm__titlebar {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  min-height: 36px;
  padding: 8px 12px;
  font-size: 24px;
  font-weight: bold;
}

.v-confirm__spacer {
  display: flex;
  height: 100%;
  flex-grow: 1;
}

.v-confirm__titlebar .v-confirm__close-icon {
  display: flex;
  align-items: center;
  min-height: 24px;
  min-width: 24px;
}

.v-confirm__message {
  overflow-wrap: break-word;
  white-space: pre-wrap;
  padding: 8px 16px;
  font-size: large;
  color: #000;
  font-weight: normal;
}

.v-confirm__actions {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 8px;
}

.v-confirm__actions > button {
  display: inline-flex;
  align-items: center;
  flex: 0 0 auto;
  justify-content: center;
  vertical-align: middle;
  height: 36px;
  min-width: 64px;
}

.v-confirm__actions > button + button {
  margin-left: 8px;
}

.v-confirm__actions > .button {
  position: relative;
  border-radius: 4px;
  font-weight: 500;
  white-space: nowrap;
  padding: 0 8px;
  font-size: 0.875rem;
  outline: none;
  appearance: none;
  border: none;
  cursor: pointer;
  color: #1867c0;
  caret-color: #1867c0;
  text-transform: uppercase;
}

.v-confirm__actions > .button::before {
  pointer-events: none;
  content: "";
  background-color: currentColor;
  border-radius: inherit;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0;
  transition: opacity 0.2s cubic-bezier(0.4, 0, 0.6, 1);
}

.v-confirm__actions > .button:hover::before {
  opacity: 0.08;
}
