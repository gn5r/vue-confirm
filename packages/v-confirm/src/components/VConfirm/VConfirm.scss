@use "./variables" as var;

@include var.theme(v-confirm) using ($theme) {
  > .v-confirm__content {
    background: map-get($theme, "background");

    .v-confirm__titlebar {
      color: map-get($theme, "text-color");

      header {
        color: inherit;
      }
    }

    .v-confirm__message {
      color: map-get($theme, "text-color");
    }
  }
}

.transition-enter-active {
  transition: var.$v-confirm-transition-enter-active;
}

.transition-leave-active {
  transition: var.$v-confirm-transition-leave-active;
}

.transition-enter,
.transition-leave-to {
  opacity: 0;
}

.v-confirm {
  position: fixed;
  z-index: 200;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: var.$v-confirm-overlay-background-color;
  display: flex;
  justify-content: center;
  align-items: center;
  outline: none;
}

.v-confirm--animated {
  > .v-confirm__content {
    animation: var.$v-confirm-animation;
  }
}

@keyframes bounce {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.05);
  }
  100% {
    transform: scale(1);
  }
}

.v-confirm__content {
  border-radius: var.$v-confirm-content-border-radius;
  box-shadow: var.$v-confirm-content-box-shadow;
  max-width: var.$v-confirm-content-max-width;
}

.v-confirm__titlebar {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  min-height: var.$v-confirm-titlebar-min-height;
  padding: var.$v-confirm-titlebar-padding;
  font-size: var.$v-confirm-titlebar-font-size;
  font-weight: var.$v-confirm-titlebar-font-weight;

  .v-confirm__close-icon {
    display: flex;
    align-items: center;
    min-height: var.$v-confirm-titlebar-icon-size;
    min-width: var.$v-confirm-titlebar-icon-size;
  }
}

.v-confirm__spacer {
  display: flex;
  height: 100%;
  flex-grow: 1;
}

.v-confirm__message {
  overflow-wrap: var.$v-confirm-message-overflow-wrap;
  white-space: var.$v-confirm-message-white-space;
  padding: var.$v-confirm-message-padding;
  font-size: var.$v-confirm-message-font-size;
  font-weight: var.$v-confirm-message-font-weight;
}

.v-confirm__actions {
  display: flex;
  align-items: center;
  padding: var.$v-confirm-actions-padding;

  // button tag
  > button {
    display: inline-flex;
    align-items: center;
    flex: 0 0 auto;
    justify-content: center;
    vertical-align: middle;
    height: var.$v-confirm-actions-button-height;
    min-width: var.$v-confirm-actions-button-min-width;
    & + button {
      margin: var.$v-confirm-actions-button-margin-size;
    }
  }

  // button class
  > .button {
    position: relative;
    border-radius: var.$v-confirm-button-border-radius;
    font-weight: var.$v-confirm-button-font-weight;
    white-space: var.$v-confirm-button-white-space;
    padding: var.$v-confirm-button-padding;
    font-size: var.$v-confirm-button-font-size;
    outline: none;
    appearance: none;
    border: none;
    cursor: pointer;
    color: var.$v-confirm-button-color;
    caret-color: var.$v-confirm-button-color;
    text-transform: uppercase;

    &::before {
      pointer-events: none;
      content: "";
      background-color: currentColor;
      border-radius: inherit;
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      opacity: 0;
      transition: opacity 0.2s cubic-bezier(0.4, 0, 0.6, 1);
    }

    &:hover::before {
      opacity: var.$v-confirm-button-hover-opacity;
    }
  }
}
