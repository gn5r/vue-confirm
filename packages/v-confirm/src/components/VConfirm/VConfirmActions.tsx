// Vue
import { defineComponent, toRef, computed } from "vue";

// Composables
import { useBackgroundColor, useTextColor } from "@/composables/color";
import { useBtnProps } from "@/composables/btn";

// Types
import type { ButtonHTMLAttributes } from "vue";

export const VConfirmActions = defineComponent({
  name: "v-confirm-footer",
  props: {
    ...useBtnProps(),
  },
  setup(props, { slots }) {
    const items = computed(() =>
      props.btns.map((btn) => {
        const { backgroundColorClasses, backgroundColorStyles } =
          useBackgroundColor(toRef(btn, "color"));
        const { textColorClasses, textColorStyles } = useTextColor(
          toRef(btn, "textColor")
        );
        const classes: string[] = [];
        if (btn.class) {
          if (Array.isArray(btn.class)) classes.push(...btn.class);
          else classes.push(btn.class);
        } else {
          classes.push("button");
        }
        return {
          attrs: {
            type: "button" as ButtonHTMLAttributes["type"],
            class: [
              ...classes,
              ...backgroundColorClasses.value,
              ...textColorClasses.value,
            ],
            style: { ...backgroundColorStyles.value, ...textColorStyles.value },
            onClick: btn.onClick,
          },
          ...btn,
        };
      })
    );

    return () => (
      <div
        class="v-confirm__actions"
        style={{ justifyContent: props.btnAlign }}
      >
        {slots.default
          ? slots.default({ items: items.value })
          : items.value.map((item) => {
              return <button {...item.attrs}>{item.text}</button>;
            })}
      </div>
    );
  },
});

export type VConfirmActions = InstanceType<typeof VConfirmActions>;
