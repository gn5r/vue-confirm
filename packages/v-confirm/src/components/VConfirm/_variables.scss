$v-confirm-transition-enter-active: opacity 0.2s ease-in;
$v-confirm-transition-leave-active: opacity 0.3s ease-out;
$v-confirm-overlay-background-color: rgba(0, 0, 0, 0.5);
$v-confirm-animation: bounce 0.2s;

$v-confirm-content-background-color: #eee;
$v-confirm-content-border-radius: 3px;
$v-confirm-content-box-shadow: 1px 2px 3px 4px rgba(12, 12, 12, 0.2);
$v-confirm-content-max-width: 95%;

$v-confirm-titlebar-min-height: 36px;
$v-confirm-titlebar-padding: 8px 12px;
$v-confirm-titlebar-font-size: 24px;
$v-confirm-titlebar-font-weight: bold;

$v-confirm-titlebar-icon-size: 24px;

$v-confirm-message-overflow-wrap: break-word;
$v-confirm-message-white-space: pre-wrap;
$v-confirm-message-padding: 8px 16px;
$v-confirm-message-font-size: large;
$v-confirm-message-font-color: #000;
$v-confirm-message-font-weight: normal;

$v-confirm-actions-padding: 8px;
$v-confirm-actions-button-height: 36px;
$v-confirm-actions-button-min-width: 64px;
$v-confirm-actions-button-margin-size: 0 0 0 8px;

$v-confirm-button-border-radius: 4px;
$v-confirm-button-font-weight: 500;
$v-confirm-button-white-space: nowrap;
$v-confirm-button-padding: 0 8px;
$v-confirm-button-font-size: 0.857rem;
$v-confirm-button-color: #1867c0;

$v-confirm-button-hover-opacity: 0.08;
