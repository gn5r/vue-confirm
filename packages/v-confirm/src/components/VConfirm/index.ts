export { VConfirm } from "./VConfirm";
export { VConfirmHeader } from "./VConfirmHeader";
export { VConfirmActions } from "./VConfirmActions";
