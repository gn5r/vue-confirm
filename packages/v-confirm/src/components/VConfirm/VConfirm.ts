// Vue
import Vue from "vue";

// Types
import { VNode, VNodeChildren } from "vue/types";
import { PropValidator } from "vue/types/options";
import { Alignment, VConfirmBtn } from "@gn5r/vue-confirm/types";

// CSS
import "./VConfirm.css";

// ClickOutside Directive
import { clickOutside } from "../../directives";

// Components
import { MdiIcon, Divider } from "../index";

// mdi close icon
import { mdiClose } from "@mdi/js";

// helper
import { getSize } from "../../utils/helper";
import { setTextColor, setBackgroundColor } from "../../utils/colorUtil";

export default Vue.extend({
  name: "v-confirm",
  directives: {
    clickOutside,
  },
  model: {
    prop: "value",
    event: "change",
  },
  props: {
    value: Boolean,
    closeable: {
      type: Boolean,
      default: true,
    },
    persistent: Boolean,
    noActionsDivider: {
      type: Boolean,
      default: false,
    },
    width: {
      type: [String, Number],
      default: "800",
    },
    title: String,
    titleColor: {
      type: String,
      default: "inherit",
    },
    titleTextColor: {
      type: String,
      default: "inherit",
    },
    closeIconColor: {
      type: String,
      default: "inherit",
    },
    message: String,
    btnAlign: {
      type: String,
      default: "end",
    } as PropValidator<Alignment>,
    btns: {
      type: Array,
      default: () => [],
    } as PropValidator<Array<VConfirmBtn>>,
  },
  data() {
    return {
      isActive: this.value,
      animated: false,
      animateTimeout: -1,
    };
  },
  computed: {
    internalValue: {
      set(val: boolean) {
        this.isActive = val;
        this.$emit("change", val);
      },
      get(): boolean {
        return this.isActive;
      },
    },
    classes(): object {
      return {
        "v-confirm": true,
        "v-confirm--animated": this.animated,
      };
    },
  },
  methods: {
    animateClick() {
      this.animated = false;
      this.$nextTick(() => {
        this.animated = true;
        window.clearTimeout(this.animateTimeout);
        this.animateTimeout = window.setTimeout(
          () => (this.animated = false),
          200
        );
      });
    },
    closeCondition(e: Event) {
      const target = e.target as HTMLElement;
      return this.isActive && (this.$refs.dialog as Element).contains(target);
    },
    onClickOutside(e: Event): void {
      if (!this.closeCondition(e)) return;

      this.$emit("click:outside", e);
      if (this.persistent) {
        this.animateClick();
      } else {
        this.internalValue = false;
      }
    },
    genDialog(): VNode {
      return this.$createElement(
        "div",
        {
          class: this.classes,
          ref: "dialog",
          directives: [{ name: "show", value: this.isActive }],
        },
        [this.genContent()]
      );
    },
    genContent(): VNode {
      return this.$createElement(
        "div",
        {
          class: "v-confirm__content",
          style: {
            width: `${getSize(this.width)}px`,
          },
          directives: [
            {
              name: "click-outside",
              value: this.onClickOutside,
            },
          ],
        },
        [
          this.genTitlebar(),
          this.genMessage(),
          this.btns.length > 0 ? this.genActions() : null,
        ]
      );
    },
    genTitlebar(): VNode {
      const children: VNodeChildren = [];
      const headerData = {
        style: {
          "background-color": "transparent",
        },
      };
      const titlebar = this.$createElement(
        "header",
        setTextColor(this.titleTextColor, headerData),
        this.title
      );
      children.push(titlebar);

      if (this.closeable) {
        const spacer = this.$createElement("div", {
          class: "v-confirm__spacer",
        });
        const closeIcon = this.$createElement(
          "div",
          { class: "v-confirm__close-icon" },
          [
            this.$createElement(
              MdiIcon,
              {
                props: {
                  color: this.closeIconColor,
                },
                on: { click: () => (this.internalValue = false) },
              },
              [mdiClose]
            ),
          ]
        );
        children.push(spacer);
        children.push(closeIcon);
      }

      const titlebarData = {
        class: {
          "v-confirm__titlebar": true,
        },
      };
      return this.$createElement(
        "div",
        setBackgroundColor(this.titleColor, titlebarData),
        children
      );
    },
    genMessage(): VNode {
      return this.$createElement(
        "div",
        {
          class: "v-confirm__message",
        },
        this.message
      );
    },
    genActions(): VNodeChildren {
      const divider = this.$createElement(Divider, { style: { margin: "0" } });
      const btns: VNodeChildren = [];
      this.btns.forEach((btn) => {
        const button = this.genActionBtn(btn);
        btns.push(button);
      });
      const actions = this.$createElement(
        "div",
        {
          class: "v-confirm__actions",
          style: {
            "justify-content": this.btnAlign,
          },
        },
        btns
      );
      return [this.noActionsDivider ? null : divider, actions];
    },
    genActionBtn(btn: VConfirmBtn): VNode {
      const text = this.$createElement("span", btn.text || null);
      const btnData = {
        class: [btn.class ? btn.class : "button"],
        attrs: {
          type: "button",
        },
        on: {
          click: btn.function,
        },
      };
      const button = this.$createElement(
        "button",
        setBackgroundColor(
          btn.backgroundColor,
          setTextColor(btn.textColor, btnData)
        ),
        [text]
      );
      return button;
    },
  },
  watch: {
    value(val: boolean): void {
      this.isActive = val;
    },
  },
  render(h): VNode {
    return h("transition", { props: { name: "transition", appear: true } }, [
      this.genDialog(),
    ]);
  },
});
