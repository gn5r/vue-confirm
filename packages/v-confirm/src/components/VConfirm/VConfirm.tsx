// Vue
import { defineComponent, ref, computed, nextTick, watch } from "vue";
import { Transition } from "vue";

// Types
import type { PropType } from "vue";
import type { Alignment, VConfirmBtn } from "@gn5r/vue-confirm/types";

// ClickOutside Directive
import { clickOutside } from "@/directives";

// Components
import { VConfirmHeader, VConfirmActions } from "@/components/VConfirm";
import { Divider } from "@/components";

// Helper
import { convertToUnit } from "@/utils/helper";
import { useThemeProps, useThemeClasses } from "@/composables/theme";

import "./VConfirm.scss";

export const VConfirm = defineComponent({
  name: "v-confirm",
  directives: {
    clickOutside,
  },
  props: {
    modelValue: Boolean,
    closeable: Boolean,
    persistent: Boolean,
    noActionsDivider: Boolean,
    width: {
      type: [String, Number],
      default: "800",
    },
    title: String,
    titleColor: {
      type: String,
      default: "inherit",
    },
    titleTextColor: {
      type: String,
      default: "inherit",
    },
    message: String,
    btnAlign: {
      type: String as PropType<Alignment>,
      default: "end",
    },
    btns: {
      type: Array as PropType<Array<VConfirmBtn>>,
      default: () => [],
    },
    hideHeader: Boolean,
    ...useThemeProps(),
  },
  emits: ["click:outside", "update:modelValue"],
  setup(props, { emit, attrs }) {
    const dialog = ref<HTMLElement>();
    const isActive = ref(props.modelValue);
    const animated = ref(false);
    const animateTimeout = ref(-1);

    const internalValue = computed({
      set: (val: boolean) => {
        isActive.value = val;
        emit("update:modelValue", val);
      },
      get: () => isActive.value,
    });
    const classes = computed(() => ({
      "v-confirm--animated": animated.value,
    }));
    const themeClasses = useThemeClasses(props);

    watch(
      () => props.modelValue,
      (val) => (isActive.value = val)
    );

    function animateClick() {
      animated.value = false;
      nextTick(() => {
        animated.value = true;
        window.clearTimeout(animateTimeout.value);
        animateTimeout.value = window.setTimeout(
          () => (animated.value = false),
          200
        );
      });
    }

    function closeCondition(e: Event) {
      const target = e.target as HTMLElement;
      return isActive.value && dialog.value?.contains(target);
    }

    function onClickOutside(e: Event) {
      if (!closeCondition(e)) return;
      emit("click:outside", e);
      if (props.persistent) animateClick();
      else internalValue.value = false;
    }

    return () => (
      <Transition name="transition" appear>
        <div
          v-show={isActive.value}
          ref={dialog}
          class={["v-confirm", classes.value, themeClasses.value]}
          {...attrs}
        >
          <div
            class="v-confirm__content"
            v-click-outside={onClickOutside}
            style={{ width: convertToUnit(props.width) }}
          >
            {props.hideHeader ? null : (
              <VConfirmHeader
                dark={props.dark}
                color={props.titleColor}
                textColor={props.titleTextColor}
                closeable={props.closeable}
                onClick={() => {
                  internalValue.value = false;
                }}
              >
                {props.title}
              </VConfirmHeader>
            )}
            <div class="v-confirm__message">{props.message}</div>
            {props.btns.length > 0 && !props.noActionsDivider && (
              <Divider dark={props.dark} style="margin: 0" />
            )}
            {props.btns.length > 0 && (
              <VConfirmActions btn-align={props.btnAlign} btns={props.btns} />
            )}
          </div>
        </div>
      </Transition>
    );
  },
});

export type VConfirm = InstanceType<typeof VConfirm>;
