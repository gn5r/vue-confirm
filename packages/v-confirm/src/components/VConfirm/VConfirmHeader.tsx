import { defineComponent, toRef } from "vue";

import { useBackgroundColor, useTextColor } from "@/composables/color";
import { useThemeProps } from "@/composables/theme";

import { MdiIcon } from "@/components";
import { mdiClose } from "@mdi/js";

export default defineComponent({
  name: "v-confirm-header",
  props: {
    color: {
      type: String,
      default: "inherit",
    },
    textColor: {
      type: String,
      default: "inherit",
    },
    closeable: Boolean,
    ...useThemeProps(),
  },
  emits: ["click"],
  setup(props, { slots, emit }) {
    const { backgroundColorClasses, backgroundColorStyles } =
      useBackgroundColor(toRef(props, "color"));
    const { textColorClasses, textColorStyles } = useTextColor(
      toRef(props, "textColor")
    );

    return () => (
      <>
        <header
          class={["v-confirm__header", backgroundColorClasses.value]}
          style={[backgroundColorStyles.value]}
        >
          <div
            class={textColorClasses.value}
            style={[{ backgroundColor: "transparent" }, textColorStyles.value]}
          >
            {slots.default?.()}
          </div>
          {props.closeable ? (
            <>
              <div class="v-confirm__spacer" />
              <div class="v-confirm__close-icon">
                <MdiIcon
                  color={props.textColor}
                  dark={props.dark}
                  onClick={() => emit("click")}
                >
                  {mdiClose}
                </MdiIcon>
              </div>
            </>
          ) : null}
        </header>
      </>
    );
  },
});
