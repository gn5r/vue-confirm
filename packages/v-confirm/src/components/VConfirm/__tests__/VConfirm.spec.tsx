import { mount } from "@vue/test-utils";
import { describe, it, expect, vi } from "vitest";
import { VConfirm } from "../VConfirm";
import type { VConfirmBtn } from "@gn5r/vue-confirm";

describe("VConfirm.tsx", () => {
  it("render with not active", () => {
    const wrapper = mount(VConfirm);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a dark component", () => {
    const wrapper = mount(VConfirm, { props: { dark: true } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a header when default", () => {
    const wrapper = mount(VConfirm);
    expect(
      wrapper.find(".v-confirm__content > .v-confirm__header").exists()
    ).toBe(true);
  });

  it("should emitted update:modelValue when clicked button on header", async () => {
    const wrapper = mount(VConfirm, {
      props: { modelValue: true, closeable: true },
    });
    await wrapper
      .find(
        ".v-confirm__content > .v-confirm__header > .v-confirm__close-icon > button"
      )
      .trigger("click");
    expect(wrapper.emitted("update:modelValue")).toBeTruthy();
  });

  it("not render a header when hideHeader is true", () => {
    const wrapper = mount(VConfirm, { props: { hideHeader: true } });
    expect(
      wrapper.find(".v-confirm__content > .v-confirm__header").exists()
    ).toBe(false);
  });

  it("component default width is 800px", () => {
    const wrapper = mount(VConfirm, {
      props: { modelValue: true },
    });
    expect(wrapper.find(".v-confirm__content").attributes().style).toContain(
      "width: 800px"
    );
  });

  it("render a component with custom width", () => {
    const wrapper = mount(VConfirm, {
      props: { modelValue: true, width: "90%" },
    });
    expect(wrapper.find(".v-confirm__content").attributes().style).toContain(
      "width: 90%"
    );
  });

  it("render a message text", () => {
    const wrapper = mount(VConfirm, {
      props: { modelValue: true, message: "message" },
    });
    expect(
      wrapper.find(".v-confirm__content > .v-confirm__message").text()
    ).toBe("message");
  });

  it("not render a divider when btns is empty", () => {
    const wrapper = mount(VConfirm, {
      props: { modelValue: true },
    });
    expect(wrapper.find(".divider").exists()).toBe(false);
  });

  it("not reneder buttons when btn is empty", () => {
    const wrapper = mount(VConfirm, {
      props: { modelValue: true },
    });
    expect(wrapper.find(".v-confirm__actions").exists()).toBe(false);
  });

  it("render a divider when the length of btns is 1 or more elements", () => {
    const btns: VConfirmBtn[] = [
      {
        text: "OK",
        function: vi.fn(),
      },
    ];
    const wrapper = mount(VConfirm, {
      props: { modelValue: true, btns: btns },
    });
    expect(wrapper.find(".divider").exists()).toBe(true);
  });

  it("not render a divider when the length of btns is 1 or more elements but noActionsDivider is true", () => {
    const btns: VConfirmBtn[] = [
      {
        text: "OK",
        function: vi.fn(),
      },
    ];
    const wrapper = mount(VConfirm, {
      props: { modelValue: true, btns: btns, noActionsDivider: true },
    });
    expect(wrapper.find(".divider").exists()).toBe(false);
  });
});
