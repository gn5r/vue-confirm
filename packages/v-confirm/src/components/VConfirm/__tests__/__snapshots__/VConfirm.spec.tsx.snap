// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VConfirm.tsx > render a dark component 1`] = `
"<transition-stub name="transition" appear="true" persisted="false" css="true">
  <div class="v-confirm dark" style="display: none;">
    <div class="v-confirm__content" style="width: 800px;">
      <header class="v-confirm__header" style="background-color: inherit;">
        <div class="" style="background-color: transparent; caret-color: inherit;">
          <!---->
        </div>
        <!---->
        <!---->
      </header>
      <div class="v-confirm__message">
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
</transition-stub>"
`;

exports[`VConfirm.tsx > render with not active 1`] = `
"<transition-stub name="transition" appear="true" persisted="false" css="true">
  <div class="v-confirm light" style="display: none;">
    <div class="v-confirm__content" style="width: 800px;">
      <header class="v-confirm__header" style="background-color: inherit;">
        <div class="" style="background-color: transparent; caret-color: inherit;">
          <!---->
        </div>
        <!---->
        <!---->
      </header>
      <div class="v-confirm__message">
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
</transition-stub>"
`;
